[Core Branding] Fix Compose Bar Text Alignment

grgia opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

Fixes alignment of compose text to match this photo:
![image](https://user-images.githubusercontent.com/38015950/207980034-13713aaf-f4c9-486c-ac88-8bc8864e318b.png)

**Current Branch:**
<img width="800" alt="image" src="https://user-images.githubusercontent.com/38015950/207999502-a17926ce-ee2e-4258-9f34-5c8ecbe82ff7.png">



### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/13595
$ https://github.com/Expensify/App/issues/14006

### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console
- [x] Go to LHN, verify the space between avatar and username is 12px
- [x] Go to a Chat
     - [x] Verify the space between avatar and username in header is 12px
     - [x] Verify the space between avatar and chat is 12px
- [x] Send 2+ Chats, verify the beginning of the text is aligned across all lines
- [x] Verify that compose bar is 40px tall in total, including the border


### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [ ] Verify that no errors appear in the JS console
- [ ] Same as tests

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->
<img width="1301" alt="image" src="https://user-images.githubusercontent.com/38015950/211684383-605ff408-ddeb-4461-83f7-30d26350c636.png">


</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->
<img width="546" alt="image" src="https://user-images.githubusercontent.com/38015950/211684718-4af508a5-f108-4daf-8023-2bde01074a75.png">
<img width="502" alt="image" src="https://user-images.githubusercontent.com/38015950/211684734-dd78abfb-0af1-44fd-8b9a-de49d1a57b06.png">

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->
<img width="520" alt="image" src="https://user-images.githubusercontent.com/38015950/211684648-b8ca4e65-0922-481c-b33d-34ce3b5d3cfd.png">
<img width="520" alt="image" src="https://user-images.githubusercontent.com/38015950/211684661-75ee6631-7f7e-4e38-b8c2-f5b9f81821f0.png">


</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->
<img width="1312" alt="image" src="https://user-images.githubusercontent.com/38015950/211684472-07346f3b-7771-40b5-aaf0-6c03cbca3751.png">


</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->
<img width="520" alt="image" src="https://user-images.githubusercontent.com/38015950/211684551-1405c643-1d4f-4091-b26b-20c9ca601644.png">
<img width="520" alt="image" src="https://user-images.githubusercontent.com/38015950/211684572-cf6c05b1-57b2-4946-807a-4c0e66bff91a.png">


</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->
<img width="502" alt="image" src="https://user-images.githubusercontent.com/38015950/211684512-104ed4d6-15ea-46d8-bbfd-87cea12b525a.png">
<img width="502" alt="image" src="https://user-images.githubusercontent.com/38015950/211684541-7a74c657-4b52-4ba0-b483-bbdb2c9253e5.png">


</details>

grgia committed:
Align compose bar text with sent text

shawnborton commented:
Tested on web but looks like we are off by just a _teeny_ bit:
<img width="739" alt="image" src="https://user-images.githubusercontent.com/2319350/207992099-e1747c92-362e-4c54-9c47-8f0eb16ceabb.png">

grgia committed:
Add 1px to right margin

shawnborton commented:
What are your thought son trying to get the little right border of the plus icon to line up with the right edge of the avatar above?

grgia commented:
How do you get the pixel overlay? Super helpful! I added 1px to the right margin

grgia commented:
Just so I'm clear, we want the right edge of the divider line to line up with the blue line, and then the left edge of the text (not including the '|' indicator) to align along the red line?

grgia committed:
Merge branch 'main' into georgia-composeFix

grgia committed:
Align divider with edge of avatar, Text with username

grgia commented:
<img width="1639" alt="image" src="https://user-images.githubusercontent.com/38015950/208002776-92734de5-39da-488d-af4b-f14d3817b7c9.png">

shawnborton commented:
Nice, this is looking pretty good! For even more alignment, I think we can make the "+" button it's own button and make it 32x32 (like the send button). Then make it have a margin right of 3px and then the border line comes right after that. This way the + is perfectly centered with the avatars above it. Something like this (added red to make it super obvious):

<img width="569" alt="image" src="https://user-images.githubusercontent.com/2319350/208326335-ec57935e-f510-4aeb-acb1-9c16d46aa74f.png">

Thoughts?

shawnborton commented:
One thing I am noticing too is that we might need to add 1 or 2 more px to the roundness of the overall compose bar. From the screenshots, it looks like it doesn't quite make a perfect pill shape:
<img width="404" alt="image" src="https://user-images.githubusercontent.com/2319350/208326385-4b9a0cc1-14df-4e30-922f-dfcef97caf9f.png">

grgia commented:
Yes, I agree with both of these suggestions! I'll get these in. One clarification- if we make the "+" button similar to the send button, the circular border (red in your pic) would still be transparent right?

shawnborton commented:
That's correct, let's still keep it transparent.

grgia commented:
<img width="600" alt="Screenshot 2022-12-20 at 6 56 44 PM" src="https://user-images.githubusercontent.com/38015950/208810029-ca971c1e-7317-4f1d-9815-d990d9703f5c.png">

Should I align the center of the expand icon with the plus or does this look good?

shawnborton commented:
This is looking good! I like the idea of center aligning the expand icon. 

So then the remaining item is the border radius of the compose box. I didn't realize it was actually 42px tall:
<img width="280" alt="image" src="https://user-images.githubusercontent.com/2319350/208810302-d73b7d03-e299-4389-a98e-4861fbbf0c31.png">


And not 40px, which is what I think I would have expected it to be. We could either adjust it so that it rests at 40px, or just bump up the border radius to at least 21px. 

My preference would be to have it at 40px because that is a standard UI component size we use elsewhere, like the size of avatars. Thoughts?

grgia committed:
Align plus/expand in compose bar, change compose height to 40px

grgia commented:
40px height + aligned expand
<img width="526" alt="Screenshot 2022-12-20 at 7 31 52 PM" src="https://user-images.githubusercontent.com/38015950/208814341-4feac162-7855-4318-8e28-ab4a3bc69175.png">
<img width="533" alt="image" src="https://user-images.githubusercontent.com/38015950/208814360-785795d5-3516-47c4-9228-10efd2b4e13b.png">

I think because the compose bar is made up of three pieces, changing border radius right now from 0 to 100 looks like this:

https://user-images.githubusercontent.com/38015950/208814536-10daaf4a-aff2-4ea6-aba1-4bc1abb99cd3.mov

shawnborton commented:
Nice, thanks for handling those. I think the issue with making the border radius anything greater than exactly half of the resting point of the compose bar height is that the corners will look too rounded when we expand the compose box. So, hence why we want 20px and hence why 20px wasn't quite cutting it when the compose box was actually 42px. 

So yeah, if we can get the compose box to start at 40px and then use 20px for border radius, we should be good to go.

grgia commented:
That makes sense! Here's where I left off last night
<img width="623" alt="image" src="https://user-images.githubusercontent.com/38015950/208968434-f1fbc256-8522-46ff-981a-f0f73a76789c.png">
<img width="622" alt="image" src="https://user-images.githubusercontent.com/38015950/208968493-36069145-ea1e-49fb-bf5c-8990309a1358.png">

shawnborton commented:
Nice! Can you code inspect to see the height of the outermost wrapper of the compose box? The wrapper that we apply the border to.

grgia committed:
Make composer total height 40px

grgia commented:
<img width="513" alt="image" src="https://user-images.githubusercontent.com/38015950/209218451-b42e2a99-f4b0-4ced-8575-f55371bcd491.png">

shawnborton commented:
Lovely, looks great! Do you have the latest version of main merged in? My eagle eye tells me that's the old version of the send icon hehe

grgia commented:
I'm impressed, you were right 😂
<img width="1130" alt="image" src="https://user-images.githubusercontent.com/38015950/209219996-a47f01e3-2d4b-47cd-9f50-65cdc242fc19.png">

grgia committed:
Merge branch 'main' into georgia-composeFix

grgia committed:
Close View component

grgia commented:
Working on getting screenshots today so we can get this merged!

grgia commented:
Okay, gonna finish up checklist and push this into review today!

grgia committed:
Merge branch 'main' into georgia-composeFix

grgia committed:
Move reused constant into variables

grgia committed:
Standardize avatar right margin to 12px in LHN, chats, etc.

grgia committed:
Align text after avatar

grgia commented:
Updated screenshots with 12px
<img width="1249" alt="image" src="https://user-images.githubusercontent.com/38015950/211648646-135cdf71-b836-43b3-8d23-33f9fdc0fd81.png">

<img width="1249" alt="image" src="https://user-images.githubusercontent.com/38015950/211648706-8605a64e-7571-4bef-a945-44fc2869c94e.png">

grgia committed:
Fix "Write something..." padding

jasperhuangg commented on src/pages/home/report/ReportActionCompose.js:
```diff
@@ -637,7 +639,7 @@ class ReportActionCompose extends React.Component {
                                     )}
                                 </AttachmentPicker>
-                                <View style={styles.textInputComposeSpacing}>
+                                <View style={[styles.textInputComposeSpacing, styles.pl6]}>
```
Let's add the `paddingLeft` styles directly to `textInputComposeSpacing`.

jasperhuangg commented on src/styles/styles.js:
```diff
@@ -350,6 +350,10 @@ const styles = {
         paddingLeft: 0,
     },
 
+    pl6: {
+        paddingLeft: 6,
+    },
+
```
Don't really need these if we follow my suggestion above, also since the paddingLeft styles aren't being used anywhere else.

shawnborton replied:
I'm also curious how this one might work with `utilities/spacing.js` in that the number after the `pl` typically denotes a multiplier which multiplies our default spacing unit of four:

<img width="320" alt="image" src="https://user-images.githubusercontent.com/2319350/211745012-2c4d21e0-bed4-48db-93f2-d5f34471454c.png">

jasperhuangg replied:
yeah, using those naming conventions `pl6` definitely isn't accurate here, it doesn't mean `paddingLeft: 6px` rather `paddingLeft: 24px`, which is why I also wanted to get rid of that class

jasperhuangg commented on src/styles/styles.js:
```diff
@@ -1739,13 +1750,13 @@ const styles = {
 
     emptyAvatar: {
-        marginRight: variables.componentSizeNormal - 24,
+        marginRight: variables.componentSizeNormal - 26,
```
Any chance we could add a comment explaining where 26 is coming from?

jasperhuangg commented on src/styles/styles.js:
```diff
@@ -1739,13 +1750,13 @@ const styles = {
 
     emptyAvatarSmall: {
-        marginRight: variables.componentSizeNormal - 28,
+        marginRight: variables.componentSizeNormal - 30,
```
Same, where is 30 coming from?

jasperhuangg reviewed (changes requested):
few small concerns, overall lookin good!

sobitneupane reviewed (commented):
>Go to LHN, verify the space between avatar and username is 12px
 Go to a Chat
 Verify the space between avatar and username in header is 12px
 Verify the space between avatar and chat is 12px
 
 @grgia Space between avatar and username is 14px in all the above cases.
 
<img width="376" alt="Screenshot 2023-01-11 at 12 50 43" src="https://user-images.githubusercontent.com/25876548/211740721-1b45c1d5-66ab-4980-89ab-9e7def4b954d.png">

grgia committed:
Use variables to control 12px spacing between avatar and chats/names

grgia commented:
Everything should be 12px now! Also, I removed that padding variable and moved dependencies into the variables folder. Should be good for re-review!

<img width="300" alt="image" src="https://user-images.githubusercontent.com/38015950/211945323-999b6047-ab26-45a2-97d1-a9166f5edf8c.png">
<img width="300" alt="image" src="https://user-images.githubusercontent.com/38015950/211945363-26fa5bfd-9b26-4718-9a75-baf1a8ce2720.png">
<img width="300" alt="image" src="https://user-images.githubusercontent.com/38015950/211945384-28ca8db8-ec02-428a-8340-825b8e020994.png">
<img width="300" alt="image" src="https://user-images.githubusercontent.com/38015950/211945216-e7d446ae-3662-4a8c-8047-60257f73c20d.png">

sobitneupane reviewed (commented):
### Screenshots/Videos
<details>
<summary>Web</summary>
<img width="466" alt="Screenshot 2023-01-12 at 10 10 00" src="https://user-images.githubusercontent.com/25876548/211976672-f4e240d9-e087-475b-8ddd-a328fbf9974e.png">

<img width="1440" alt="Screenshot 2023-01-12 at 10 15 32" src="https://user-images.githubusercontent.com/25876548/211976653-c603512e-cdab-4100-8daf-be90a11a6f06.png">

<img width="680" alt="Screenshot 2023-01-12 at 10 21 49" src="https://user-images.githubusercontent.com/25876548/211977253-ec7a7cd7-f6eb-4701-90be-5c694346358b.png">

</details>

<details>
<summary>Mobile Web - Chrome</summary>
<img width="974" alt="Screenshot 2023-01-12 at 12 00 17" src="https://user-images.githubusercontent.com/25876548/211992408-fecc19cc-476f-41c9-93da-71f2e6e6a21c.png">

<img width="333" alt="Screenshot 2023-01-12 at 12 02 49" src="https://user-images.githubusercontent.com/25876548/211992371-e219cf17-5727-4279-9f55-9342e4596c81.png">

<img width="974" alt="Screenshot 2023-01-12 at 12 02 24" src="https://user-images.githubusercontent.com/25876548/211992394-95182c0e-fdf9-463f-8af7-3168fd7ef7da.png">

</details>

<details>
<summary>Mobile Web - Safari</summary>

<img width="1341" alt="Screenshot 2023-01-12 at 11 06 53" src="https://user-images.githubusercontent.com/25876548/211985248-f1697213-fd18-42e1-bd48-aa374a1bb44a.png">

<img width="280" alt="Screenshot 2023-01-12 at 11 17 27" src="https://user-images.githubusercontent.com/25876548/211985412-a7cf8f04-edf9-4177-ab9c-b0b038d54a4f.png">

<img width="1341" alt="Screenshot 2023-01-12 at 11 15 56" src="https://user-images.githubusercontent.com/25876548/211985243-a8d4ce0a-2072-46c1-ae5a-14a19f940a0a.png">
</details>

<details>
<summary>Desktop</summary>

<img width="1198" alt="Screenshot 2023-01-12 at 10 58 10" src="https://user-images.githubusercontent.com/25876548/211983307-7462797c-5046-4640-8194-4ad0b1588465.png">
<img width="1198" alt="Screenshot 2023-01-12 at 10 57 55" src="https://user-images.githubusercontent.com/25876548/211983316-204a328f-8abd-4aa0-b5ce-921d3e9b4076.png">
<img width="1198" alt="Screenshot 2023-01-12 at 11 00 58" src="https://user-images.githubusercontent.com/25876548/211983288-8751b5d4-47f4-4986-a506-997ebcc58c34.png">
<img width="1198" alt="Screenshot 2023-01-12 at 10 58 20" src="https://user-images.githubusercontent.com/25876548/211983301-f7a5243f-0cec-462b-bcce-2ba5d4ef50c8.png">
</details>

<details>
<summary>iOS</summary>
<img src="https://user-images.githubusercontent.com/25876548/211986440-ea16c233-9a4b-40d4-adb6-57afe37521bb.png" width="350" />

<img width="350" alt="Screenshot 2023-01-12 at 11 24 35" src="https://user-images.githubusercontent.com/25876548/211986454-f3c98e12-2ee0-44a0-b137-9ee64be96f47.png">

<img width = "350" src="https://user-images.githubusercontent.com/25876548/211986432-e4bba2ed-844f-473c-a8d3-df5b7377b16e.png" />

</details>

<details>
<summary>Android</summary>
<img width="332" src="https://user-images.githubusercontent.com/25876548/211991062-5b20f3ea-a0bc-4e96-b0a8-7c14c0657b59.png" />

<img width="332" alt="Screenshot 2023-01-12 at 11 55 19" src="https://user-images.githubusercontent.com/25876548/211991036-6c1abc9f-6e7c-4c41-9817-2ed9c364c966.png">

</details>

sobitneupane reviewed (approved):
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

shawnborton commented:
Looks great to me! I smacked the "Ready to build" label thinking it might trigger some testable builds? Will keep an eye there. Otherwise feel free to merge this one.

shawnborton commented:
I'm gonna go ahead and merge this one, not sure why the builds didn't work but 🤷

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/3912408220) to staging by @shawnborton in version: 1.2.54-1 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/3933816087) to production by @AndrewGable in version: 1.2.54-2 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
